managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/LedgerMetadataUtils.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.mledger.impl;
+
+import com.google.common.collect.ImmutableMap;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/**
+ * Utilities for managing BookKeeper Ledgers custom metadata.
+ */
+public final class LedgerMetadataUtils {
+
+    private static final String METADATA_PROPERTY_APPLICATION = "application";
+    private static final byte[] METADATA_PROPERTY_APPLICATION_PULSAR
+            = "pulsar".getBytes(StandardCharsets.UTF_8);
+
+    private static final String METADATA_PROPERTY_COMPONENT = "component";
+    private static final byte[] METADATA_PROPERTY_COMPONENT_MANAGED_LEDGER
+            = "managed-ledger".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] METADATA_PROPERTY_COMPONENT_COMPACTED_LEDGER
+            = "compacted-ledger".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] METADATA_PROPERTY_COMPONENT_SCHEMA
+            = "schema".getBytes(StandardCharsets.UTF_8);
+
+    private static final String METADATA_PROPERTY_MANAGED_LEDGER_NAME = "psr/managed-ledger";
```

Comment:
I'd prefer to keep `pulsar` vs `psr` since the 3 bytes saved won't justify the time spent thinking what's "psr" 😄

Reply:
@merlimat @sijie I don't know if it is worth to add such prefix, as we already have two preexisting keys without it for  compacted topics.
I am open to any suggestion

Reply:
"application" and "component" are general keys that can be used for other applications. the other fields are pulsar specific, which I think they would be prefixed with "pulsar". because beside pulsar components, we are also using dlog for storing pulsar functions and table service for pulsar state. so the ledgers can be used for different purpose, having prefix will make things clear and avoid property conflicts.

with that being said, 

"compactedTopic" should be "pulsar/compactedTopic",
"compactedTo" should be "pulsar/compactedTo",
"schemaId" should be "pulsar/schemaId"